.github/workflows/CD.yml

```diff
@@ -68,3 +39,105 @@ jobs:
           aws s3 cp $rpm_artifact s3://artifacts.opendistroforelasticsearch.amazon.com/downloads/rpms/opendistro-knn/
           aws s3 cp $deb_artifact s3://artifacts.opendistroforelasticsearch.amazon.com/downloads/debs/opendistro-knn/
           aws cloudfront create-invalidation --distribution-id E1VG5HMIWI4SA2 --paths "/*"
+
+  library-build-and-ship-zip-and-rpm:
+    name: Build k-NN JNI Library for Zip and RPM
+    runs-on: ubuntu-latest
+    container:
+      image: centos:7
+    strategy:
+      matrix:
+        java: [14]
+        compiler: [g++]
+    steps:
+      - name: Checkout k-NN
+        uses: actions/checkout@v1
+        with:
+          submodules: true
+
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: us-east-1
+
+      - name: Setup Java ${{ matrix.java }}
+        uses: actions/setup-java@v1
+        with:
+          java-version: ${{ matrix.java }}
+
+      - name: Install dependencies
+        run: |
+          yum update -y
+          yum install -y cmake rpm-build gcc-c++ make
+
+      - name: Build and ship library RPM and ZIP
+        env:
+          CXX: ${{ matrix.compiler }}
+        run: |
+          cd jni
+          sed -i 's/-march=native/-march=x86-64/g' external/nmslib/similarity_search/CMakeLists.txt
+          cmake .
+          make package
+
+          cd packages
+          folder_name=`ls ./*.rpm | sed 's|\(.*\)\..*|\1|'`
+          zip_name=$folder_name".zip"
+          mkdir $folder_name
+          cp ../release/*.so $folder_name
+          zip -r $zip_name $folder_name/*
+          cd ..
+
+          zip_artifact=`ls packages/*.zip`
+          rpm_artifact=`ls packages/*.rpm`
+
+          aws s3 cp $zip_artifact s3://artifacts.opendistroforelasticsearch.amazon.com/downloads/opendistro-libs/opendistro-knnlib/
+          aws s3 cp $rpm_artifact s3://artifacts.opendistroforelasticsearch.amazon.com/downloads/rpms/opendistro-knnlib/
+          aws cloudfront create-invalidation --distribution-id E1VG5HMIWI4SA2 --paths "/*"
+
+  library-build-and-ship-deb:
```

Comment:
As we discussed @jmazanec15 since CentOS have the lowest version of glibc, it would make sense to compile all distros of knnlib (zip/deb/rpm) in CentOS.

Reply:
Yes, I believe that makes sense. Will test artifacts built there on other systems.